pkg/filesystem/path/windows_parser.go

```diff
@@ -0,0 +1,78 @@
+package path
+
+import (
+	"strings"
+	"unicode"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type windowsParser struct {
+	absolute    bool
+	driveLetter rune
+	path        string
+}
+
+// NewWindowsParser creates a Parser for Windows paths that can be used in Resolve.
+func NewWindowsParser(path string) (Parser, error) {
+	// Windows paths are generally passed to system calls that
+	// accept C strings. There is no way these can accept null
+	// bytes.
+	driveLetter := rune(0)
+	absolute := false
+
+	if strings.ContainsRune(path, '\x00') {
+		return nil, status.Error(codes.InvalidArgument, "Path contains a null byte")
+	}
+
+	path = strings.Replace(path, "/", "\\", -1)
+	// C:\
+	if len(path) >= 3 && unicode.IsLetter(rune(path[0])) && path[1] == ':' && path[2] == '\\' {
+		driveLetter = rune(path[0])
+		path = path[3:]
+	}
+
+	if len(path) >= 1 && path[0] == '\\' {
+		absolute = true
+	}
```

Comment:
My recommendation is to do this as part of ParseScope(). That prevents the need for storing all of these things separately in `windowsParser`.

Reply:
Good idea, that makes it much neater! Less state to hang on to and the ParseScope() function is improved too.